packages/hash/integration/src/tests/model/knowledge/user.model.test.ts

```diff
@@ -82,7 +85,7 @@ describe("User model class", () => {
 
     expect(fetchedUser).not.toBeNull();
 
-    expect(fetchedUser).toEqual(createdUser);
+    expect(fetchedUser?.entityId).toEqual(createdUser?.entityId);
```

Comment:
Why have you changed this?

Reply:
Thanks for flagging! Deep inequality between entities was failing which interfered with prior work in the PR, but I did not intend on committing this workaround and forgot to look into it.

Have looked into it now, and the reason was the Graph API returns `requiredLinks: undefined` instead of `links: []`. This means the deep inequality started failing because of how I implemented `links` in `generateWorkspaceEntityTypeSchema`. Have addressed this in https://github.com/hashintel/hash/pull/1068/commits/9959b525a2ac34737d83c7e747ede3ea3506889b, and removed this diff.

Reply:
I've also made the change for the `required` field in https://github.com/hashintel/hash/pull/1068/commits/5a13445fb06065d8c502be72edbf59540f0ffb97